projects/storefrontlib/shared/components/captcha/config/captcha-api-config.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, Type } from '@angular/core';
+import { Config } from '@spartacus/core';
+import { CaptchaProvider } from '../captcha.model';
+
+@Injectable({
+  providedIn: 'root',
+  useExisting: Config,
+})
+export abstract class CaptchaApiConfig {
+  apiUrl?: string;
```

Comment:
I think having `apiUrl` and `fields` can still be relevant only to specific captchas (google's in this case). Lets try to be as agnostic as we can at this generic captcha layer.
We can move these to a `GoogleRecaptcha2` options object that can apply them as `captchaOptions`.